addon/mixins/route.js

```diff
@@ -283,17 +278,61 @@ export default Ember.Mixin.create({
 
    @method updateInfinityModel
    @param {Ember.Enumerable} newObjects The new objects to add to the model
-   @return {Ember.Array} returns the updated infinity model
+   @return {DS.RecordArray} returns the updated infinity model
    */
   updateInfinityModel(newObjects) {
```

Comment:
we should also add a test to ensure that we preserve the functionality you're concerned about... it's not currently tested